pkg/report/coverageprofile.go

```diff
@@ -0,0 +1,306 @@
+		TotalViolationLines: violationLines,
+		ViolationSections: []*ViolationSection{
+			{
+				ViolationLines: violationLines,
+				StartLine:      change.Sections[0].StartLine,
+				EndLine:        change.Sections[0].EndLine,
+				Contents:       change.Sections[0].Contents,
+			},
+		},
+	}
+
+	return coverageProfile
+}
+
+// generateCoverageProfileWithModifyMode generates for modify file
+func generateCoverageProfileWithModifyMode(profile *cover.Profile, change *gittool.Change) *CoverageProfile {
+
+	blocks := profile.Blocks
+	sort.Slice(blocks, func(i, j int) bool {
+		return blocks[i].StartLine < blocks[j].StartLine
+	})
+
+	var total, covered int64
+	var totalViolationLines []int
+	var violationSections []*ViolationSection
+
+	// for each section contents, find each line in profile block and judge it
+	for _, section := range change.Sections {
+
+		var violationLines []int
+		for lineNo := section.StartLine; lineNo <= section.EndLine; lineNo++ {
+			block := binarySeachForProfileBlock(blocks, 0, len(blocks)-1, lineNo)
+			if block == nil {
+				continue
+			}
+
+			// check line?
+
+			total++
+			if block.Count > 0 {
+				covered++
+			} else {
+				violationLines = append(violationLines, lineNo)
+			}
+
+		}
+
+		if len(violationLines) != 0 {
+			violationSections = append(violationSections, &ViolationSection{
+				StartLine:      section.StartLine,
+				EndLine:        section.EndLine,
+				Contents:       section.Contents,
+				ViolationLines: violationLines,
+			})
+
+			totalViolationLines = append(totalViolationLines, violationLines...)
+		}
+
+	}
+
+	if total == 0 {
+		return nil
+	}
+
+	coveredPercent := int(float64(covered) / float64(total) * 100)
+
+	return &CoverageProfile{
+		FileName:            change.FileName,
+		TotalLines:          int(total),
+		CoveredLines:        int(covered),
+		CoveragePercent:     coveredPercent,
+		TotalViolationLines: totalViolationLines,
+		ViolationSections:   violationSections,
+	}
+}
+
+func binarySeachForProfileBlock(blocks []cover.ProfileBlock, left int, right int, lineNo int) *cover.ProfileBlock {
+
+	var mid int
+	for left <= right {
+		mid = (left + right) >> 1
+		// current line is in this block
+		if blocks[mid].StartLine <= lineNo && blocks[mid].EndLine >= lineNo {
+			return &blocks[mid]
+		}
+		if blocks[mid].StartLine > lineNo {
+			right = mid - 1
+		} else {
+			left = mid + 1
+		}
+	}
+
+	return nil
+}
+
+// findChange find the expected change by file name
+func findChange(profile *cover.Profile, changes []*gittool.Change) *gittool.Change {
+	for _, change := range changes {
+		if strings.HasSuffix(profile.FileName, change.FileName) {
+			return change
+		}
+	}
+	return nil
+}
+
+// sortLines returns the lines in increasing order.
+func sortLines(m map[int]bool) []int {
+	var lines []int
+	for k := range m {
+		lines = append(lines, k)
+	}
+	sort.Ints(lines)
+	return lines
+}
+
+// returns the maximum value between two int values.
+func maxInt(a, b int) int {
```

Comment:
seems like this helper function is not being used?

Reply:
Removed.